src/services/identity/LaunchesService.ts

```diff
@@ -146,6 +165,84 @@ export class LaunchesService {
     return ok(siteId)
   }
 
+  getDNSRecords = (
+    repoName: string
+  ): ResultAsync<DnsResultsForSite, AmplifyError | SiteLaunchError> => {
+    let dnsRecords: DNSRecord[] = []
+    let siteUrl: string
+    return ResultAsync.fromPromise(this.getSiteId(repoName), () =>
+      createErrorAndLog(AmplifyError, `Failed to get siteId for ${repoName}`)
+    )
+      .andThen((siteId) => {
+        if (siteId.isOk()) {
+          return ResultAsync.fromPromise(
+            this.launchesRepository.findOne({
+              where: { siteId: siteId.value },
+            }),
+            () =>
+              createErrorAndLog(
+                AmplifyError,
+                `Failed to get launch record for ${repoName}`
+              )
+          )
+        }
+        return errAsync(
+          createErrorAndLog(
+            AmplifyError,
+            `Failed to get launch record for ${repoName}`
+          )
+        )
+      })
+      .andThen((launchRecord) => {
+        if (!launchRecord) {
+          return errAsync(
+            createErrorAndLog(SiteLaunchError, "Failed to get launch record")
+          )
+        }
+        siteUrl = launchRecord.primaryDomainSource
+        dnsRecords.push({
+          source: launchRecord.primaryDomainSource,
+          target: launchRecord.primaryDomainTarget,
+          type: "CNAME",
+        })
+        dnsRecords.push({
+          source: launchRecord.domainValidationSource,
+          target: launchRecord.domainValidationTarget,
+          type: "CNAME",
+        })
+        return okAsync(launchRecord)
+      })
+      .andThen((launchRecord) =>
+        ResultAsync.fromPromise(
+          this.redirectionsRepository.findOne({
+            where: { launchId: launchRecord.id },
+          }),
+          () =>
+            // There are legitimate cases where this error is thrown (ie the site has no redirection record)
+            new AmplifyError(`Failed to get redirection record for ${repoName}`)
+        ).andThen((redirectionRecord) => {
```

Comment:
nit: is it possible to flatten this out instead of chaining multiple levels?

Reply:
was thinking that we will primarily use the isOk isErr constructs and write the error handling in usual JS/TS way of if, else instead of the method chaining constructs from neverthrow

Reply:
hmm am quite confused here ah, how can I consume an async neverthrow without using chaining like this? 
I think it also might be beneficial if we clarified on slack whether we want to encourage/discourage chaining as a team!

Reply:
i think we have to consume `ResultAsync` using `andThen` - this is by design because `await` will turn it into a `Promise` and escape from our error handling patterns

Reply:
shouldn't be a blocker if the code is working - we can iterate on this as a ticket next time.

Reply:
side note is that when we write as a `Promise<Result>` and we do `isOk + await`, we can resolve both of those using `andThen` ._.